### ISSUE TITLE
[Input]: Not extract the TypeScript type of nested schema

### ISSUE THREAD

**Issue by hungify** [fabian-hiller/valibot#34]
Hi and thank you for your time on it
With the simple code below or [Playground](https://tsplay.dev/WK3zzW)
Why do types in `name` still contain `v.ObjectInput`?

```
import * as v from "valibot"; // 0.74 kB

const loginSchema = v.object({
  email: v.string([
    v.minLength(1, "Please enter your email."),
    v.email("The email address is badly formatted."),
  ]),
  password: v.string([
    v.minLength(1, "Please enter your password."),
    v.minLength(8, "You password must have 8 characters or more."),
  ]),
  name: v.object({
    first: v.string([v.minLength(1, "Please enter your first name.")]),
    last: v.string([v.minLength(1, "Please enter your last name.")]),
  }),
});

type Login = v.Input<typeof loginSchema>;
//   ^?
```
![image](https://github.com/fabian-hiller/valibot/assets/78008509/5f971b62-a362-48fb-a2cf-47b15b3b0557)

*I think, all examples on documentation should use a wildcard import to easy for the reader and less confusing*

**Comment by yamiteru** [comment]
If I understand it correctly `Input`/`Output` do not infer nested schemas (https://github.com/fabian-hiller/valibot/blob/e00461daafbb1e636c4b1846f80fe41cd3881753/library/src/types.ts#L55C12-L55C12).

**Comment by fabian-hiller** [comment]
I suspect that this is due to TypeScript and I have only indirect influence on it. But the type itself is correct. TypeScript is just displaying it weirdly. I will investigate it anyway. This issue it related to #10.

> I think, all examples on documentation should use a wildcard import to easy for the reader and less confusing

Why do you think wildcard imports are easier to understand? Feel free to explain in more detail. Currently I still prefer single imports.

> If I understand it correctly Input/Output do not infer nested schemas

This is not correct, at least if I have not made a mistake. The `types` property always contains the final input and output type of a schema. You can assign a complex type to a variable and then check if nested properties are suggested.

**Comment by yamiteru** [comment]
I believe in these cases it's good to unwrap the type using this little handy thing:

```ts
export type Pretty<$Type> = {
	[$Key in keyof $Type]: $Type[$Key];
} & {};
```

**Comment by fabian-hiller** [comment]
Should be fixed in v0.8.0.
